src/core/ext/transport/chaotic_good/client/chaotic_good_connector.cc

```diff
@@ -0,0 +1,329 @@
+    if (is_shutdown_) {
+      auto error = GRPC_ERROR_CREATE("connector shutdown");
+      MaybeNotify(DEBUG_LOCATION, notify, error);
+      return;
+    }
+    result_ = result;
+  }
+  args_ = std::move(args);
+  notify_ = notify;
+  resolved_addr_ = EventEngine::ResolvedAddress(
+      reinterpret_cast<const sockaddr*>(args.address->addr), args.address->len);
+  GPR_ASSERT(resolved_addr_.value().address() != nullptr);
+  Ref().release();  // Ref held by OnHandshakeDone().
+  grpc_event_engine::experimental::EventEngine::OnConnectCallback on_connect =
+      [this](absl::StatusOr<std::unique_ptr<EventEngine::Endpoint>> endpoint) {
+        if (!endpoint.ok() || handshake_mgr_ == nullptr) {
+          auto error = GRPC_ERROR_CREATE("connect endpoint failed");
+          MaybeNotify(DEBUG_LOCATION, notify_, error);
+          return;
+        }
+        ExecCtx exec_ctx;
+        handshake_mgr_->DoHandshake(
+            grpc_event_engine_endpoint_create(std::move(endpoint.value())),
+            channel_args_, args_.deadline, nullptr /* acceptor */,
+            OnHandshakeDone, this);
+      };
+  event_engine_->Connect(
+      std::move(on_connect), *resolved_addr_, ee_config_,
+      memory_quota_->CreateMemoryAllocator("control_endpoint_connection"),
+      timeout_);
+}
+
+void ChaoticGoodConnector::OnHandshakeDone(void* arg, grpc_error_handle error) {
+  auto* args = static_cast<HandshakerArgs*>(arg);
+  ChaoticGoodConnector* self =
+      static_cast<ChaoticGoodConnector*>(args->user_data);
+  {
+    MutexLock lock(&self->mu_);
+    if (!error.ok() || self->is_shutdown_) {
+      if (error.ok()) {
+        error = GRPC_ERROR_CREATE("connector shutdown");
+        if (args->endpoint != nullptr) {
+          grpc_endpoint_shutdown(args->endpoint, error);
+          grpc_endpoint_destroy(args->endpoint);
+          grpc_slice_buffer_destroy(args->read_buffer);
+          gpr_free(args->read_buffer);
+        }
+      }
+      self->result_->Reset();
+      MaybeNotify(DEBUG_LOCATION, self->notify_, error);
+      return;
+    }
+  }
+  if (args->endpoint != nullptr) {
+    GPR_ASSERT(grpc_event_engine::experimental::grpc_is_event_engine_endpoint(
+        args->endpoint));
+    self->control_endpoint_ = std::make_shared<PromiseEndpoint>(
+        grpc_event_engine::experimental::
+            grpc_take_wrapped_event_engine_endpoint(args->endpoint),
+        SliceBuffer());
+    self->connect_activity_ = ReceiveSettingsFrame(self);
+  } else {
+    // Handshaking succeeded but there is no endpoint.
+    {
+      MutexLock lock(&self->mu_);
+      self->result_->Reset();
+      auto error = GRPC_ERROR_CREATE("handshake complete with empty endpoint.");
+      MaybeNotify(DEBUG_LOCATION, self->notify_, error);
+    }
+  }
+  self->handshake_mgr_.reset();
+}
+
+ActivityPtr ChaoticGoodConnector::ReceiveSettingsFrame(
+    ChaoticGoodConnector* self) {
+  GPR_ASSERT(self->control_endpoint_ != nullptr);
+  auto read_setting_frames = TrySeq(
```

Comment:
Please take a look at #35400 and separate the stages into individual functions - this `TrySeq` is too long and too nested to comprehend readily